frontend/lib/search/location_button.dart

```diff
@@ -32,21 +36,17 @@ class _LocationButtonState extends State<LocationButton> {
 
   @override
   Widget build(BuildContext context) {
-    if (_locationStatus == LocationRequestStatus.requestSuccessful) {
-      return Container();
-    }
     final settings = Provider.of<SettingsModel>(context);
     final t = context.t;
-    if (settings.locationFeatureEnabled) {
+    if (widget.sortingMode == SortingMode.alphabetically) {
       return Container(
```

Comment:
since there is lots of duplicated code, i think its nicer to create ab FAB widget and use there all the static parameters and pass the only text, icon and onPress fkt.
I think you should also add an icon for alphabetically sorting to keep the look and feel equal
https://api.flutter.dev/flutter/material/Icons-class.html